odoo/addons/base/models/res_bank.py

```diff
@@ -52,6 +52,19 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
                 domain = ['&'] + domain
         bank_ids = self._search(domain + args, limit=limit, access_rights_uid=name_get_uid)
         return self.browse(bank_ids).name_get()
+        
+    @api.onchange('country')
+    def _onchange_country_id(self):
+        for state in self.env['res.country.state'].search([('name', 'ilike', self.state.name)]):
+            if state.country_id == self.country:
+                self.state = state
+        if self.country != self.state.country_id:
```

Comment:
nonsense... this just needs to be simply:
``` python
def _onchange_country_id(self):
    if self.state and self.country != self.state.country_id:
        self.state = False
```

Reply:
That would have been a hefty performance cost into production hehe.

Reply:
of course, if the user selected 'Barcelona (spain)' and then select Venezuela in the country field you want to ditch the state you've previously chosen. Otherwise, it's inconsistent

Reply:
What it is doing is that it replaces with Barcelona (VE). One less click for the user, useful because the field is removed without a warning. This makes the changes on name_get kind of useless but I thing this chunk is really useful.

Reply:
And still, the name_get helps the user to not have to input the country, only the state because the state will be filled automatically if he sees that he selects the right (state,country).